Remotion/Validation/UnitTests/Implementation/ObjectValidatorFactoryTest.cs

```diff
@@ -15,29 +15,77 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using Moq;
 using NUnit.Framework;
+using Remotion.Reflection;
+using Remotion.Validation.Implementation;
+using Remotion.Validation.Rules;
+using Remotion.Validation.UnitTests.TestDomain;
+using Remotion.Validation.UnitTests.TestDomain.Validators;
+using Remotion.Validation.UnitTests.TestHelpers;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Implementation
 {
   [TestFixture]
   public class ObjectValidatorFactoryTest
   {
-    [Ignore("RM-5906")]
     [Test]
-    public void Create ()
+    public void Create_ReturnsValidator ()
     {
+      var fakeCustomerValidator = new FakeCustomerValidator();
+      ObjectValidationRuleInitializationParameters actualInitializationParameters = null;
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactoryFunc = parameters =>
+      {
+        actualInitializationParameters = parameters;
+        return fakeCustomerValidator;
+      };
+
+      var validatedType = TypeAdapter.Create(typeof(Person));
+
+      var validationMessageFactoryMock = new Mock<IValidationMessageFactory>(MockBehavior.Strict);
+      validationMessageFactoryMock
+          .Setup(_ => _.CreateValidationMessageForObjectValidator(fakeCustomerValidator, validatedType))
+          .Returns(new InvariantValidationMessage("test"));
+
+      var result = ObjectValidatorFactory.Create<IObjectValidator>(validatedType, validatorFactoryFunc, validationMessageFactoryMock.Object);
+
+      Assert.That(result, Is.SameAs(fakeCustomerValidator));
+      Assert.That(actualInitializationParameters.ValidationMessage.ToString(), Is.EqualTo("test"));
```

Comment:
```suggestion
      Assert.That(result, Is.SameAs(fakeCustomerValidator));
      Assert.That(actualInitializationParameters, Is.Not.Null);
      Assert.That(actualInitializationParameters.ValidationMessage.ToString(), Is.EqualTo("test"));
```

This is actually a good practice in these cases becasue it gives a clear error in case the call didn't happen / the parameter was left null. Otherwise, we get a NullRefEx in the build and that's more annoying to figure out.